File: app/src/main/java/ai/elimu/analytics/rest/StoryBookLearningEventService.java
package ai.elimu.analytics.rest;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Multipart;
import retrofit2.http.PUT;
import retrofit2.http.Part;

public interface StoryBookLearningEventService {

    @Multipart
    @PUT("analytics/storybook-learning-events/csv")
    Call<ResponseBody> uploadCsvFile(@Part("csvFile") RequestBody csvFileRequestBody);
}

File: app/src/main/java/ai/elimu/analytics/rest/StoryBookLearningEventService.kt
package ai.elimu.analytics.rest

import okhttp3.MultipartBody
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part

interface StoryBookLearningEventService: UploadService {
    @Multipart
    @POST("analytics/storybook-learning-events/csv")
    override fun uploadCsvFile(@Part part: MultipartBody.Part): Call<ResponseBody?>
}
